### PR TITLE
feat: [M3-7169, M3-7193, M3-7211, M3-7255, M3-7258] - VPC UX feedback

### PR DESCRIPTION
**Pull request by hana-linode**
## Description 📝
Address UX feedback for VPC

## Changes  🔄
- Updated VPC empty landing text
- New VPC icon
- Minor UI changes to Assign Linode to Subnet drawer
- Updated tooltip text for the Auto-assign a VPC IPv4 address checkbox
- Minor text UI changes to VPC details page

## Preview 📷
| VPC empty landing before  | VPC empty landing after   |
| ------- | ------- |
| image | vpc empty after |

| VPC side/create menu icon before  | VPC side/create menu after   |
| ------- | ------- |
| image | vpc icon after |

| Assign Linode to Subnet drawer before  | Assign Linode to Subnet drawer after   |
| ------- | ------- |
| Screenshot 2023-10-23 at 5 26 22 PM |  Screenshot 2023-10-23 at 5 27 24 PM |

| VPC detail Safari before  | VPC detail Safari after   |
| ------- | ------- |
| image | image |

### Prerequisites
(How to setup test environment)
- Point to the dev env and ensure your account has vpc customer tags

### Verification steps 
Verify the ticket changes requested and check these flows:
- VPC empty landing (set line 87 to true in `VPCLanding`)
- VPC side menu and create menu icons
- Assign Linode to Subnet drawer
- VPC details page (changes are more noticeable in Safari)

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

**Comment by mjac0bs** [comment]
@hana-linode Relevant unit test failure: 
image

**Review by mjac0bs — COMMENTED** [review]
Confirmed:

Icon has been replaced and looks good in light and dark mode + all screen sizes within:
- [x] Sidenav
- [x] Create Menu
- [x] VPC Empty State

Empty state landing:
- [x] Header changed from "VPCs" to "Virtual Private Cloud (VPC)"
- [x] Removed period at the end of "Create a private and isolated network." to be consistent with other empty state pages.
- [x] Added byline "Enable cloud resources to privately and securely communicate with each other, the internet, and other private networks." under the "Create a private and isolated network" text.

VPC Details:

- [x] Updated Subnets, Linodes, Region, VPC ID, Created, Updated and Description labels are less bold and consistent with the Linodes details summary.
- [x] "Subnets (X)" header font-size is smaller.

Assign Linode to Subnet Drawer:
- [x] The "Auto-assign a VPC IPv4 address for this Linode" label is middle aligned in relation to the checkbox.
- [x] The "Linodes Assigned to Subnet (0)" text font-weight is less bold and consistent with the h1 header.
- [?] The "Download List of Assigned Linodes (.csv)" link is not be displayed until a Linode has been assigned to the subnet.
   - I'm not totally sure what's going on with this last one, but I could only use the mocks to confirm it displays only once a linode is assigned because I can't seem to get the linodes assigned to my subnet to load. I noticed the nested table has an infinite loading spinner. (I did see this working yesterday when testing another PR + have both tags on my account, so I'm not sure what has changed.)
Screenshot 2023-10-24 at 2 08 34 PM

The “Auto-assign a VPC IPv4 address for this Linode” checkbox tooltip is to “A VPC IPv4 is the private IP address for this Linode in the VPC.” in:
- Create Linode -- This was listed in the internal ticket, but I don't think it applies?
- [x] Assign Linode to Subnet
- [x] Linode details Edit Config
- [x] Linode details  Add Config

Unrelated to this PR, here were some other minor UX things I found, for the VPC team to decide if they're worth addressing:
- An edge case with VPC description - if a user were to put in one very long word, it overflows the container. This is also noticeable as the screen size gets smaller -- the words don't have to be as long. (This does also happen on a stack script's details page and maybe elsewhere around the app.)
Screenshot 2023-10-24 at 2 01 16 PM
- On small screens on the VPC Detail page, the subnet searchbar text gets larger as the screen size decreases (I think this is common throughout the app), but it also cuts off the placeholder text so "or id" is obscured.
Screenshot 2023-10-24 at 2 14 52 PM (2)

**Comment by mjac0bs** [comment]
Just had a look at the CI run and it looks like `vpc-landing-page.spec.ts` and `vpc-details-page.spec.ts` need the copy updates too - those are relevant failures.

**Comment by hana-linode** [comment]
@mjac0bs merged develop in and the nested table should be loading again. I tried looking into the overflow issue but was not able to find an easy solution, so I created M3-7331 to track that

**Review by mjac0bs — APPROVED** [review]
Tests are passing and nested table looks good again!

File: packages/manager/src/features/VPCs/VPCDetail/SubnetAssignLinodesDrawer.tsx [hunk 1]
```diff
@@ -371,16 +375,29 @@ export const SubnetAssignLinodesDrawer = (
...
-            'A range of non-internet facing IP used in an internal network'
-          }
-          checked={autoAssignIPv4}
-          disabled={userCannotAssignLinodes}
-          onChange={handleAutoAssignIPv4Change}
-          sx={{ marginLeft: `2px`, marginTop: `8px` }}
-          text={'Auto-assign a VPC IPv4 address for this Linode'}
-        />
+        <Box alignItems="center" display="flex" flexDirection="row">
+          <FormControlLabel
```
Comment (coliu-akamai) [inline packages/manager/src/features/VPCs/VPCDetail/SubnetAssignLinodesDrawer.tsx:379]: could we add the `disabled={userCannotAssignLinodes}` to this form control too?

image

**Review by coliu-akamai — APPROVED** [review]
🎉  Awesome, looks great!! Confirmed the following:

✅ M3-7169 (empty landing state)
- verified VPC landing text
- verified icon change

✅ M3-7193 (subnet assign linodes drawer)
- verified assign linode to subnet drawer changes - csv + alignment
  - (I think the font weight issue is something that only appears on safari and is being handled in another pr by @abailly-akamai? Will double check)
image

✅ M3-7211 (vpc details)
- confirmed font size changes!

✅ M3-7255 (vpc icon)
- confirmed icon changes!

✅ M3-7258
- confirmed copy changes for tooltip in create linode, assign linode to subnet, add/edit config flows

✅ confirmed tests still pass!

**Comment by coliu-akamai** [comment]
Confirmed @abailly-akamai's pr #9843 fixes the font weight on safari for the Subnet Assign Linodes drawer!
image

File: packages/manager/src/features/OneClickApps/oneClickApps.ts
Comment (dwiley-akamai) [inline packages/manager/src/features/OneClickApps/oneClickApps.ts:1]: How'd this get pulled in 🧐

File: packages/manager/src/features/VPCs/utils.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import type { Config, Subnet } from '@linode/api-v4';
 
+export const VPC_LABEL = 'Virtual Private Cloud (VPC)';
```
Comment (dwiley-akamai) [inline packages/manager/src/features/VPCs/utils.ts:3]: We should move this to `/VPCs/constants.ts`

File: packages/manager/src/features/VPCs/VPCDetail/SubnetAssignLinodesDrawer.tsx [hunk 3]
```diff
@@ -371,16 +375,30 @@ export const SubnetAssignLinodesDrawer = (
...
+                Auto-assign a VPC IPv4 address for this Linode
+              </Typography>
+            }
+            data-testid="vpc-ipv4-checkbox"
+            disabled={userCannotAssignLinodes}
+            sx={{ marginRight: 0 }}
+          />
+          <TooltipIcon
+            text={
+              'A VPC IPv4 is the private IP address for this Linode in the VPC.'
```
Comment (dwiley-akamai) [inline packages/manager/src/features/VPCs/VPCDetail/SubnetAssignLinodesDrawer.tsx:397]: We should store this in a variable in `/VPCs/constants.ts` and use it here & in `VPCPanel.tsx` (ditto for "Auto-assign a VPC IPv4 address for this Linode" actually)

**Review by dwiley-akamai — APPROVED** [review]
LGTM, left a few minor comments!

VPC empty landing ✅
VPC side menu and create menu icons ✅
Assign Linode to Subnet drawer ✅
VPC details page ✅

**Review by cliu-akamai — APPROVED** [review]

File: packages/manager/src/features/OneClickApps/oneClickApps.ts
Comment (hana-linode) [reply packages/manager/src/features/OneClickApps/oneClickApps.ts:1]: Prob the auto linter 🤔

**Pull request merged**